shell/browser/ui/file_dialog_linux.cc

```diff
@@ -0,0 +1,260 @@
+// Copyright (c) 2024 Microsoft, Gmbh.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include <memory>
+#include <string>
+
+#include "base/files/file_util.h"
+#include "base/functional/bind.h"
+#include "base/functional/callback.h"
+#include "base/memory/raw_ptr.h"
+#include "base/memory/raw_ptr_exclusion.h"
+#include "base/run_loop.h"
+#include "base/strings/string_util.h"
+#include "base/strings/utf_string_conversions.h"
+#include "shell/browser/javascript_environment.h"
+#include "shell/browser/native_window_views.h"
+#include "shell/browser/ui/file_dialog.h"
+#include "shell/browser/ui/select_file_policy.h"
+#include "shell/common/gin_converters/callback_converter.h"
+#include "shell/common/gin_converters/file_path_converter.h"
+#include "ui/shell_dialogs/select_file_dialog.h"
+#include "ui/shell_dialogs/selected_file_info.h"
+
+namespace file_dialog {
+
+DialogSettings::DialogSettings() = default;
+DialogSettings::DialogSettings(const DialogSettings&) = default;
+DialogSettings::~DialogSettings() = default;
+
+namespace {
+
+ui::SelectFileDialog::Type GetDialogType(int properties) {
+  if (properties & OPEN_DIALOG_OPEN_FILE) {
+    return ui::SelectFileDialog::SELECT_OPEN_FILE;
+  } else if (properties & OPEN_DIALOG_OPEN_DIRECTORY) {
+    return ui::SelectFileDialog::SELECT_FOLDER;
+  } else if (properties & OPEN_DIALOG_MULTI_SELECTIONS) {
+    return ui::SelectFileDialog::SELECT_OPEN_MULTI_FILE;
+  }
+
+  return ui::SelectFileDialog::SELECT_OPEN_FILE;
```

Comment:
_(opinion)_ `else` after a return is a [little weird](https://clang.llvm.org/extra/clang-tidy/checks/readability/else-after-return.html).

```suggestion
  if (properties & OPEN_DIALOG_OPEN_FILE)
    return ui::SelectFileDialog::SELECT_OPEN_FILE;
    
  if (properties & OPEN_DIALOG_OPEN_DIRECTORY)
    return ui::SelectFileDialog::SELECT_FOLDER;
    
  if (properties & OPEN_DIALOG_MULTI_SELECTIONS)
    return ui::SelectFileDialog::SELECT_OPEN_MULTI_FILE;
    
  return ui::SelectFileDialog::SELECT_OPEN_FILE;
```